api/agent/runner_client.go

```diff
@@ -208,11 +208,11 @@ func (r *gRPCRunner) TryExec(ctx context.Context, call pool.RunnerCall) (bool, e
 	}
 }
 
-func sendToRunner(ctx context.Context, protocolClient pb.RunnerProtocol_EngageClient, call pool.RunnerCall) {
+func sendToRunner(ctx context.Context, protocolClient pb.RunnerProtocol_EngageClient, runnerAddress string, call pool.RunnerCall) {
 	bodyReader := call.RequestBody()
 	writeBuffer := make([]byte, MaxDataChunk)
 
-	log := common.Logger(ctx)
+	log := common.Logger(ctx).WithField("runner_addr", runnerAddress)
```

Comment:
i guess for a not-well-configured install this is necessary? most logging systems will give this to us tho just by way of sending the logs in. I suspect it would be nice to hang this off of the root logger somewhere in the server package with something like hostname to get this for all logs? can slide it in for debugging I guess, it just seems like it will be a duplicate at some point (maybe not?)

Reply:
I'm not sure that will work. These lines are being logged on the lb nodes (grpc client), not the runner nodes (grpc server) so the hostname/ip address would always be the same and correspond to the lb. I'd like us to log what runner we are connected to rather than what host is acting as client here.

Reply:
ah, that makes sense, thanks for explanation